### ISSUE TITLE
After turn on ATC, couldn't connect web site from client side 

### ISSUE THREAD

**Issue by JohnnyWuMigme** [facebook/augmented-traffic-control#144]
Hi Sir, 

I'm new on using this great tool. I had installed the ATC tool on my Ubuntu VM, what I've done is setup a wireless USB device as wifi-ap, so my mobile devices / laptop can connect with and access ATC control panel. I also setup the DHCP service. Before I turn on the ATC from control panel, the network connection is fine, I can access web from browser without problem. However, once I turn on ATC, I can still ping to internet without problem, and the response time also increases (like real 3G network), but once I try to open any web page, the action would always timeout and fail. 

Below are the curl information:
```
➜  ~  curl www.yahoo.com
<HTML>
<HEAD>
<TITLE>Error</TITLE>
</HEAD>

<BODY BGCOLOR="white" FGCOLOR="black">
<!-- status code : 301 -->
<!-- Error: GET -->
<!-- host machine: ir6.fp.tp2.yahoo.com -->
<!-- timestamp: 1439377398.000 -->
<!-- url: http://www.yahoo.com/-->
<H1>Error</H1>
<HR>

<FONT FACE="Helvetica,Arial"><B>
Description: Could not process this "GET" request.
</B></FONT>
<HR>
</BODY>
```
Is there any setting I should check? I have no idea where might be wrong. It looks like the iptables drop packets on port 80 or 443. Does anyone encounter similar situation? Thanks for your kind response!

**Comment by chantra** [comment]
@JohnnyWuMigme this is working as expected and you  are actually getting a reply from yahoo.


Also, running the same curl on my own laptop without atc in the middle as the same outcome.

Use: `curl -v www.yahoo.com` and you will see that you get a redirect to https://www.yahoo.com

I will close this for now as it does not seem to be an issue. Feel free to open a new task if anything comes up.

**Comment by JohnnyWuMigme** [comment]
Hi Chantra,

I check with curl -v and below are the comparison info:

Without ATC:

➜  ~  curl -v www.yahoo.com
* Rebuilt URL to: www.yahoo.com/
* Hostname was NOT found in DNS cache
*   Trying 116.214.12.74...
* Connected to www.yahoo.com (116.214.12.74) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.37.1
> Host: www.yahoo.com
> Accept: */*
> 
< HTTP/1.1 301 Redirect
< Date: Thu, 13 Aug 2015 01:17:06 GMT
< Via: http/1.1 ir1.fp.tp2.yahoo.com (ApacheTrafficServer)
* Server ATS is not blacklisted
< Server: ATS
< Location: https://www.yahoo.com/
< Content-Type: text/html
< Content-Language: en
< Cache-Control: no-store, no-cache
< Connection: keep-alive
< Content-Length: 374
< 
<HTML>
<HEAD>
<TITLE>Error</TITLE>
</HEAD>

<BODY BGCOLOR="white" FGCOLOR="black">





<H1>Error</H1>
<HR>

<FONT FACE="Helvetica,Arial"><B>
Description: Could not process this "GET" request.
</B></FONT>
<HR>
</BODY>
* Connection #0 to host www.yahoo.com left intact

With ATC:

➜  ~  curl -v www.yahoo.com
* Rebuilt URL to: www.yahoo.com/
* Hostname was NOT found in DNS cache
*   Trying 116.214.12.74...
*   Trying 2406:2000:ec:601::1009...
* Immediate connect fail for 2406:2000:ec:601::1009: No route to host
*   Trying 2406:2000:ec:601::1009...
* Immediate connect fail for 2406:2000:ec:601::1009: No route to host
* Connected to www.yahoo.com (116.214.12.74) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.37.1
> Host: www.yahoo.com
> Accept: */*
> 
< HTTP/1.1 301 Redirect
< Date: Thu, 13 Aug 2015 01:17:53 GMT
< Via: http/1.1 ir6.fp.tp2.yahoo.com (ApacheTrafficServer)
* Server ATS is not blacklisted
< Server: ATS
< Location: https://www.yahoo.com/
< Content-Type: text/html
< Content-Language: en
< Cache-Control: no-store, no-cache
< Connection: keep-alive
< Content-Length: 374
< 
<HTML>
<HEAD>
<TITLE>Error</TITLE>
</HEAD>

<BODY BGCOLOR="white" FGCOLOR="black">





<H1>Error</H1>
<HR>

<FONT FACE="Helvetica,Arial"><B>
Description: Could not process this "GET" request.
</B></FONT>
<HR>
</BODY>
* Connection #0 to host www.yahoo.com left intact


We can see after turn on ATC, there's "no route to host" message displayed. Think that's why I can't get access with website after turn on ATC? Do you have any suggestion to dig deeper?

Thanks!

**Comment by chantra** [comment]
@JohnnyWuMigme 
Are you testing on the with and without ATC on the same router?

Anyhow, in both cases, you are getting a connection to yahoo.com (yahoo is replying with some HTML).

Also, the 'no route to host' is only happening over IPv6. This is a routing issue, not an ATC issue and I suspect that even if ATC was not running on your router, you would get the same error.

ATC does not handle routing, it sits on top and just tell the linux kernel to shape specific traffic.
Moreover, ATC does not support ipv6, so essentially your IPv6 is not shaped by ATC.

**Comment by JohnnyWuMigme** [comment]
OK, I gave up last time. I give a try today and find the workaround for me. Before I used VM installing Ubuntu on my mac pro, and I connect an external wireless usb device as the wifi ap, and I use network sharing for the VM network (eth0 in Ubuntu), so the mac pro on board wifi chip is the wan side. I guess this would cause some problem, today I connect a thunderbot - ethernet adaptor and disable the mac wifi, then everything's running normal! Hope if someone has same problem as me, this workaround could help you. Thanks!

**Comment by JohnnyWuMigme** [comment]
Hi, 
root cause should be the way that vm share network with local os.
before my vm network settings is "Share with my Mac", after I choose "Bridge Networking" then everything's running as expected. Thanks!!!

**Comment by chantra** [comment]
:+1:

**Comment by mgjnana** [comment]
@JohnnyWuMigme, i encounters a similar problem, but i can't see..how can i setup Bridge Networking? wish your answer, thank U!

**Comment by mgjnana** [comment]
@chantra  @JohnnyWuMigme , 
i encounters a similar problem, but i can't see..how can i setup Bridge Networking? wish your answer, thank U! wait on line...

**Comment by chantra** [comment]
@mgjnana what virtualization system are you using? Individual software providers usually have docs about it. Quick googling yields https://www.virtualbox.org/manual/ch06.html#network_bridged for virtual box, and vmware: https://www.vmware.com/support/ws4/doc/network_bridged_ws.html

**Comment by mgjnana** [comment]
@chantra  vmware, and i tried every way, no luckly ,I found vm cant't found  wireless card,finally , I installed ubuntu（14.04 LTS） on my commputer solved this problem. My job is to build a network test environment, I was watching atc_demo_ui now, JSX ,but thank you all the same.  : )
